### ISSUE TITLE
feat(model): admit retained Q2_K embedding rows to Vulkan Qwen3.8 sequence prefill

### ISSUE THREAD

**Issue by anthony-chaudhary** [anthony-chaudhary/fak#12105]
# feat(model): admit retained Q2_K embedding rows to Vulkan Qwen3.8 sequence prefill

```routing
lane: model
paths: ["internal/model/qwen35_hal.go", "internal/model/embedding_q2k.go", "internal/model/*qwen35*test.go"]
expected_steps: 3
```

Parent: #11572. Builds on completed #11942. Blocks #11959 and #11963. Related safety guard: #12091.

## Current evidence

Commit `f0a02030c` retains the Q2_K token embedding and provides bounded native row gathering, but `tryQwen35SequencePrefill` still explicitly declines when `Q2KEmbedding != nil`. Materializing the 248,320 x 5,120 embedding as F32 would require about 5.08 GB and can exceed a Vulkan single-resource cap.

## Deliverable

1. Gather/dequantize only the prompt's requested Q2_K embedding rows into a bounded token panel.
2. Feed that panel to the existing Vulkan Qwen sequence-prefill request without allocating the full F32 table.
3. Preserve first/middle/last/repeated-row CPU-oracle parity, request lifetime, and exact token order.
4. Missing native capability must be explicit; native-performance receipts cannot silently qualify a per-token CPU fallback.

## Scope fences

- No new quantization format, multi-buffer full-table upload, QKNorm implementation, or speculative decoding.

## Acceptance

- A focused fail-before/pass-after regression proves Q2_K models enter sequence prefill with bounded memory and match the existing row oracle.
- `go test -v ./internal/model -run "TestQ2KEmbedding|TestQwen35SequencePrefill.*Q2K" -count=1` exits 0.
- A sanctioned Strix Halo smoke records the native Vulkan sequence path, memory envelope, model/source identity, quality, and zero implicit fallback.

## Done condition

Retained Q2_K embeddings participate in bounded whole-sequence prefill without full-table F32 expansion.

**Comment by anthony-chaudhary** [comment]
Landed on `main` at `2e4ed9e7a` (implementation landing `f8bc48c1d`; physically tested source `65e42a81fc7c80e6f55ffbc4ca258db2fb56dc26`).

What changed:
- retained Q2_K input embeddings now gather/dequantize only `[tokens, hidden]` rows in exact first/middle/last/repeated order;
- CUDA and Vulkan share an explicit, mutually exclusive row-panel request contract;
- missing row capability is an explicit non-qualifying fallback, wrong identity is typed, and latent whole-table HAL expansion is refused;
- the uploaded panel has a single caller owner and exact one-time free.

Verification:
- `go test -v ./internal/model -run "TestQ2KEmbedding|TestQwen35SequencePrefill.*Q2K" -count=1` PASS;
- `go test ./internal/compute -count=1` PASS;
- `go vet ./internal/model ./internal/compute` PASS;
- full 5-gate boundary audit PASS.

Physical RADV STRIX_HALO witness (`model.test` SHA256 `23f50d7dbef71a5c44553e52df5f493e9d0a6c8df9d58c24ca03dcd1d69dcd59`): native whole-sequence path, 4,096-byte row panel vs 5,085,593,600-byte production f32 table and 4,294,967,292-byte device cap, logits cosine 1.0, exact argmax, and zero H2D/D2H bytes after panel admission. The active `Qwen3.8-27B-Q4_K_M` service was neither restarted nor reloaded.

Receipt: https://github.com/anthony-chaudhary/fak/blob/main/docs/benchmarks/receipts/strix-vulkan-qwen35-q2k-embedding-rows-20260908.json

Scope fence: this closes the bounded operation-level deliverable. Exact full-checkpoint generation remains unqualified and tracked by #11959.
